Follow the linter’s accessibility suggestions

fregante opened the pull request:
This PR enables the rules that were soft-enabled globally and fixes some of the inline exclusions.

fregante commented on src/components/documentBuilder/preview/DocumentPreview.tsx:
```diff
@@ -63,7 +63,7 @@ const DocumentPreview = ({
 
   return (
-    // eslint-disable-next-line jsx-a11y/mouse-events-have-key-events
+    // eslint-disable-next-line jsx-a11y/mouse-events-have-key-events, jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions -- TODO
```
I'll leave this as a TOFIX since the component is probably a bit complex.

fregante commented on src/components/formBuilder/FormPreviewFieldTemplate.tsx:
```diff
@@ -15,8 +15,6 @@
  */
 
-/* eslint-disable jsx-a11y/click-events-have-key-events */
-/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
```
This is just arbitrarily ignoring the linter, which means either the suggestion isn't applicable to us (and therefore we should disable it globally), but that's likely not the case.

fregante commented on src/components/jsonTree/treeHooks.tsx:
```diff
@@ -35,9 +36,9 @@ export function useLabelRenderer() {
-          className={styles.copyPath}
+        <Button
+          variant="text"
+          className={[styles.copyPath, "p-0"].join(" ")}
```
<img width="270" alt="Screen Shot 2" src="https://user-images.githubusercontent.com/1402241/148359666-003c009b-eca0-4319-92de-3a8890640d39.png">

twschiller replied:
In react there's a utility for combining classnames which also cleanly support conditional class names:

```
import cx from "classnames";
cx(styles.copyPath, "p-0")
```

fregante commented on src/devTools/editor/fields/TemplateWidget.tsx:
```diff
@@ -59,18 +59,18 @@ const TemplateWidget: React.FC<TemplateWidgetProps> = ({
-            <a
+            <Button
+              variant="link"
+              size="sm"
```
Matches the style of the "shortcuts" below. We could alternatively  fix both to use the new `btn-reset-link` style I added, which makes them all look like real textual links.

<img width="537" alt="Screen Shot 1" src="https://user-images.githubusercontent.com/1402241/148359706-a64ce67a-8921-479f-bc39-e346094556fc.png">

fregante commented on src/devTools/editor/tabs/editTab/editorNode/EditorNode.tsx:
```diff
@@ -93,6 +93,7 @@ const EditorNode: React.FC<EditorNodeProps> = ({
 
   return (
     // Use our own custom style here, not bootstrap
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
```
```suggestion
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events -- TODO
```

fregante commented on src/options/pages/SetupPage.tsx:
```diff
@@ -129,9 +129,13 @@ const SetupPage: React.FunctionComponent = () => {
+                  >
                     local-only mode
-                  </a>
+                  </Button>
```
![gif](https://user-images.githubusercontent.com/1402241/148359966-cae66fa3-91be-48e6-8a60-8719877c48f8.gif)

fregante commented on src/options/pages/services/SharedServicesCard.tsx:
```diff
@@ -92,25 +92,16 @@ const SharedServicesCard: React.FunctionComponent<OwnProps> = ({
-              create a PixieBrix account
-            </a>
-          </span>
-        )}
```
I don’t think this is ever possible? The `serviceUrl` uses the default when the field is empty. For this to happen you'd probably have to explicitly use the env `SERVICE_URL=`, which does not happen on production.

fregante commented on src/vendors/overrides.scss:
```diff
@@ -18,3 +18,20 @@
+    color: #0056b3;
+    text-decoration: underline;
+  }
+}
```
I didn't find a file to place globally-reusable styles and I'd like one. It seems that this file is used everywhere. 

Alternatively I suppose I could extend `Button` and create a `TextButton` JSX component with this style instead.

twschiller replied:
> Alternatively I suppose I could extend Button and create a TextButton JSX component with this style instead.

I'd lean toward creating a component (in the components directory) and adding it to the storybook

fregante replied:
Added `LinkButton`

fregante commented on src/devTools/editor/tabs/editTab/editorNode/EditorNode.tsx:
```diff
@@ -93,6 +93,7 @@ const EditorNode: React.FC<EditorNodeProps> = ({
 
   return (
     // Use our own custom style here, not bootstrap
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events -- TODO
```
This has been fixed in https://github.com/pixiebrix/pixiebrix-extension/pull/2300 🎉 

I'll fix the merge conflict once either one is merged

twschiller reviewed (approved):
See comments and merge whenever ready

fregante commented:
All done. I also updated the "Shortcuts" links spacing I mentioned before. Updated shots:
<img width="790" alt="Screen Shot 4" src="https://user-images.githubusercontent.com/1402241/148495758-845a553a-5f01-419f-9df7-68d711911b2e.png">
<img width="493" alt="Screen Shot 3" src="https://user-images.githubusercontent.com/1402241/148495761-2aeb3729-250a-49e1-abe3-e02a77a39df4.png">


<img width="735" alt="Screen Shot 5" src="https://user-images.githubusercontent.com/1402241/148495735-7cd308f0-fdb5-4b0c-a498-6ac49058e117.png">
